x/asset/keeper/restriction.go

```diff
@@ -0,0 +1,54 @@
+package keeper
+
+import (
+	errorsmod "cosmossdk.io/errors"
+	"github.com/realiotech/realio-network/x/asset/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type RestrictionChecker interface {
+	IsAllow(ctx sdk.Context, tokenId string, sender string) (bool, error)
+}
+
+func (k Keeper) AssetSendRestriction(ctx sdk.Context, fromAddr, toAddr sdk.AccAddress, amt sdk.Coins) (newToAddr sdk.AccAddress, err error) {
+	newToAddr = toAddr
+	err = nil
+
+	// if no checker exist allow all sender
+	if len(k.RestrictionChecker) == 0 {
+		return newToAddr, nil
+	}
+
+	checker := k.RestrictionChecker[0]
```

Comment:
There is a comment in `x/asset/keeper/keeper.go` that we should not support more than 1 `RestrictionChecker`.  However, the code seems to allow more than one anyway (both `x/asset/keeper/keeper.go` and `x/asset/keeper/restriction.go` have the conditional check `len(k.RestrictionChecker) == 0` instead of something like `len(k.RestrictionChecker) != 0` for instance.

This line `checker := k.RestrictionChecker[0]` will ignore any `RestrictionChecker` after the first one registered.  If multiple exist, how can we be sure the first is the correct one to use?  It seems the code should either enforce only one `RestrictionChecker`, or this code block should iterate all `RestrictionCheckers` in this list.